src/selections-factory.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState, useSelections } from '@nebula.js/stardust';
+
+export function selectCell(selections, cell) {
```

Comment:
getting `Cognitive Complexity of 10 (exceeds 5 allowed)` here. Sure, there is a lot of logic in this func. Could split this up, but not sure it adds any value? opinions?

Reply:
I changed the rules so 8 is fine. But i think we shouldn't get more complex than this.

Reply:
think this is fine